metagraph/src/cli/load/load_graph.cpp

```diff
@@ -71,5 +71,47 @@ std::shared_ptr<DeBruijnGraph> load_critical_dbg(const std::string &filename) {
     exit(1);
 }
 
+template <class Graph>
+std::shared_ptr<CanonicalDBG> primary_to_canonical(std::shared_ptr<Graph> graph,
+                                                   size_t cache_size) {
+    if (graph->get_mode() != DeBruijnGraph::PRIMARY) {
+        logger->error("Only primary mode graphs can be wrapped into canonical mode.");
+        exit(1);
+    }
+
+    logger->trace("Primary graph wrapped into canonical");
+    return std::make_shared<CanonicalDBG>(graph, cache_size);
+}
+
+template std::shared_ptr<CanonicalDBG> primary_to_canonical(std::shared_ptr<const DeBruijnGraph>, size_t);
+template std::shared_ptr<CanonicalDBG> primary_to_canonical(std::shared_ptr<DeBruijnGraph>, size_t);
+
+std::shared_ptr<const DeBruijnGraph> make_cached_graph(const DeBruijnGraph &graph,
+                                                       const Config &config,
+                                                       size_t cache_size) {
+    std::shared_ptr<const DeBruijnGraph> ret_graph(std::shared_ptr<const DeBruijnGraph>{}, &graph);
+
+    // if alignment in both directions is not required, then there's no need to cache
+    if (ret_graph->get_mode() != DeBruijnGraph::CANONICAL && config.align_only_forwards)
+        return ret_graph;
```

Comment:
1. So how is this function called `make_cached_graph` and it may return a graph which is not cached? Did you not ask yourself this question? If you keep it as is, you should rename it to `prepare_alignment_graph` or something like that.

2. It might be ok to just always cache. Is there any slowdown or downside of doing it if it's not required?

Reply:
I cleaned this up a bit, so now we don't need this function anymore

Reply:
Also, it's not always a good idea to have a cached view. The internal caches use mutexes, so if you're using a single base graph in a multi-threaded setting, then having the cache can slow things down.

This is why I have a separate cached view for each thread